src/librustc_mir/interpret/memory.rs

```diff
@@ -50,7 +54,9 @@ pub struct Memory<'a, 'mir, 'tcx: 'a + 'mir, M: Machine<'a, 'mir, 'tcx>> {
     /// deallocation.  When an allocation is not found here, it is a
     /// static and looked up in the `tcx` for read access.  Writing to
     /// a static creates a copy here, in the machine.
-    alloc_map: FxHashMap<AllocId, (MemoryKind<M::MemoryKinds>, Allocation)>,
+    /// We have this in a RefCell to be able to cache when a new allocation
+    /// for a static has to be created on a read access.
```

Comment:
The comment seems internally inconsistent. It claims that writing creates a copy but also that reading creates a copy..?